### PR TITLE
Add ability to unlink an approved speaker from a talk #JOINDIN-647

### PR DESCRIPTION
**Pull request by railto**
Work in progess, feedback appreciated

### CODE REVIEW

File: app/src/Talk/TalkController.php [hunk 1]
```diff
@@ -444,6 +446,34 @@ public function reportComment($eventSlug, $talkSlug, $commentHash)
         $this->application->redirect($url);
     }
 
+    public function unlinkSpeaker($eventSlug, $talkSlug, $username)
+    {
+        if (!isset($_SESSION['user'])) {
+            $thisUrl = $this->application->urlFor('talk', ['eventSlug' => $eventSlug, 'talkSlug' => $talkSlug]);
```
Comment (heiglandreas) [inline app/src/Talk/TalkController.php]: I'd move this out of the if-clause to be able to use it as redirection URL after the action is finished.

File: app/src/Talk/TalkController.php [hunk 2]
```diff
@@ -444,6 +446,34 @@ public function reportComment($eventSlug, $talkSlug, $commentHash)
...
+
+        $userApi = $this->getUserApi();
+        $user = $userApi->getUserByUsername($username);
+        $userId = $user->getId();
+
+        try {
+            // TODO: Make API call to unlink speaker from talk
+        } catch (Exception $e) {
+            // TODO: Handle exception
+        }
```
Comment (heiglandreas) [inline app/src/Talk/TalkController.php]: What do you do after the action is finished?

I'd redirect to the talk-page (which should then have one speaker less)

Comment (railto) [inline app/src/Talk/TalkController.php]: That's my plan, Just missed putting in a todo for it :)

File: app/src/Talk/TalkController.php (hunk 1 above)
Comment (railto) [inline app/src/Talk/TalkController.php]: Good point, updating now.

**Comment by railto** [comment]
In commit/6196023 I mentioned that it's not showing flash message, just realised it is, but not at top of page.  My thought would be that this is outside the scope of this issue.  I can remove the flash message completely or leave it there until we decide if should be moved or not.

**Comment by railto** [comment]
@liam-wiltshire @heiglandreas Could one of you take a final look at this when you get time please.

**Pull request merged**